pkg/reconciler/revision/resources/deploy.go

```diff
@@ -194,12 +186,50 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingC
 	if observabilityConfig.EnableVarLogCollection {
 		podSpec.Volumes = append(podSpec.Volumes, internalVolume)
 	}
+	return podSpec, nil
+}
 
-	if autoscalerConfig.EnableGracefulScaledown {
-		podSpec.Volumes = append(podSpec.Volumes, labelVolume)
+func appendContainer(old []corev1.Container, new corev1.Container) []corev1.Container {
+	for key := range old {
+		if old[key].Name == new.Name {
+			// need to check with more negetive scenarios
+			return old
+		}
 	}
+	return append(old, new)
+}
 
-	return podSpec, nil
+func makeContainer(userContainer *corev1.Container, rev *v1alpha1.Revision) *corev1.Container {
+	userContainer.VolumeMounts = append(userContainer.VolumeMounts, varLogVolumeMount)
+	userContainer.Lifecycle = userLifecycle
+	userContainer.Env = append(userContainer.Env, getKnativeEnvVar(rev)...)
+	// Explicitly disable stdin and tty allocation
+	userContainer.Stdin = false
+	userContainer.TTY = false
+	if userContainer.TerminationMessagePolicy == "" {
+		userContainer.TerminationMessagePolicy = corev1.TerminationMessageFallbackToLogsOnError
+	}
+	return userContainer
+}
+
+func createServingContainer(rev *v1alpha1.Revision) *corev1.Container {
```

Comment:
Shall we make this take a `container` as well? That'd make it's behavior clearer with the conditionals above and it'd be consistent with `makeContainer`

Also: Do you need to return a pointer? You're just dereferencing it above anyway.